execute_tests.sh

```diff
@@ -136,38 +145,53 @@ echo "+-------------------- ANSIBLE INVENTORY --------------------+"
 # Run molecule converge and verify
 set +e # allow test stages to return errors
 for TEST in "${MOLECULES[@]}" ; do
-    moleculerize --output "$TEST/molecule/default/molecule.yml" dynamic_inventory.json
-    pushd "$TEST"
-    repo_uri=$(git remote -v | awk '/fetch/{print $2}')
-    echo "TESTING: $repo_uri at SHA $(git rev-parse HEAD)"
-
-    # Capture the molecule test repo in the environment so "pytest-rpc" can record it.
-    export MOLECULE_TEST_REPO=$(echo ${repo_uri} | rev | cut -d'/' -f1 - | rev | cut -d. -f1)
-
-    # Capture the SHA of the tests we are executing
-    export MOLECULE_GIT_COMMIT=$(git rev-parse HEAD)
-
-    # Execute the converge step
-    if [[ "${SKIP_CONVERGE}" = false ]]; then
-        molecule --debug converge
-    else
-        echo "Skipping converge step!"
-    fi
-
-    if [[ $? -ne 0 ]] && RC=$?; then  # do not run tests if converge fails
-        echo "CONVERGE: Failure in $(basename ${TEST}), verify step being skipped"
-        continue
-    fi
-
-    # Execute the verify step
-    if [[ "${SKIP_VERIFY}" = false ]]; then
-        molecule --debug verify
-    else
-        echo "Skipping verify step!"
-    fi
-
-    [[ $? -ne 0 ]] && RC=$?  # record non-zero exit code
-    popd
+    # Introspect all scenarios from given molecule
+    ALL_SCENARIOS=()
+    SCENARIO_PATHS=($TEST/molecule/*)
+    for path in "${SCENARIO_PATHS[@]}"; do
+        ALL_SCENARIOS+=("${path##*/}")
```

Comment:
What is this? Just curious on how it works.

Reply:
`"${STR##*/}"` returns the basepath for a given path string.
```
$ STR="/path/to/scenarios/foo"
$ echo ${STR##*/}
foo
```


Ref: https://devhints.io/bash#parameter-expansions

Reply:
![why didn't I know about this!?](https://media.giphy.com/media/xT5LMHGhUg0FkktVBe/giphy.gif)